src/common/com/intellij/plugins/haxe/build/FieldWrapper.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2000-2013 JetBrains s.r.o.
+ * Copyright 2014-2016 AS3Boyan
+ * Copyright 2014-2014 Elias Ku
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intellij.plugins.haxe.build;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+
+/**
+ * Access a field using reflection.
+ *
+ * Typically, we are using this for backward code compatibility, so that we
+ * can get around the compile-time requirement for missing fields to be
+ * present (e.g. have syntactically correct code when the underlying
+ * sources have changed).
+ *
+ * Created by ebishton on 1/19/16.
+ */
+public class FieldWrapper<FieldType> {
```

Comment:
We are using reflection to deal with missing fields.  This allows us to look up a field that should be there in one version but isn't in another.  That way we can wrap the code with if (IdeaTarget.IS_VERSION_15_COMPATIBLE){} and not have compilation errors.